dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/ProgramRuleEngine.java

```diff
@@ -52,32 +52,19 @@
 import org.hisp.dhis.rules.models.*;
 import org.hisp.dhis.user.CurrentUserService;
 import org.hisp.dhis.user.UserAuthorityGroup;
-import org.springframework.stereotype.Component;
 import org.springframework.transaction.annotation.Transactional;
 
-import lombok.extern.slf4j.Slf4j;
-
 /**
  * Created by zubair@dhis2.org on 11.10.17.
  */
 @Slf4j
 @Transactional( readOnly = true )
```

Comment:
Why do we need to open a transaction here? It seems that all the dependencies open their own TX?
Also, I noticed that the class `DefaultProgramRuleService` uses a class level `@Transactional` annotation, which open a read/write TX also for get only method. I know it's outside the scope of this PR, but perhaps we can fix it.